testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -152,14 +152,14 @@ object ServiceTest {
       result match {
         case asyncResult: Future[_] =>
           import testServer.executionContext
-          asyncResult.onComplete { _ =>
+          asyncResult.map { theResult =>
             testServer.stop()
-          }
+            theResult
+          }.asInstanceOf[R]
```

Comment:
hmmm, this is wrong: `map` would only run when the `asyncResult: Future` is successful.